nwapp/utils/notifier.js

```diff
@@ -28,14 +36,85 @@ function playNotificationSound() {
   audio = null;
 }
 
+
+// Cache lookup
+var urlFilePathMap = new Map();
+function getFileAtUrl(url) {
+  return new Promise(function(resolve, reject) {
+    var cachedFilePath = urlFilePathMap.get(url);
+    if(cachedFilePath) {
+      resolve(cachedFilePath);
+    }
+    else {
+      setTimeout(function() {
+        reject(new Error('Took to long to request file', url));
+      }, 750);
+
+      var tempFileStream = temp.createWriteStream();
+      tempFileStream
+        .on('finish', function() {
+          var tempFilePath = tempFileStream.path;
+          // Store it on our map as a cache lookup
+          urlFilePathMap.set(url, tempFilePath);
+          resolve(tempFilePath);
+        })
+        .on('error', function(err) {
+          reject(err);
+        });
+
+      request(url)
+        .on('error', function(err) {
+          reject(err);
+        })
+        .pipe(tempFileStream);
+    }
+  });
+}
+
 module.exports = function (spec) {
   if (!settings.showNotifications) return;
-  playNotificationSound();
 
-  var notification = new window.Notification(spec.title, {
-    body: spec.message,
-    icon: getNotificationImageForUri(spec.link)
-  });
+  var avatarImageUrl = getFullUrlNotificationImageForUri(spec.link);
+
+  getFileAtUrl(avatarImageUrl)
```

Comment:
@trevorah `nw.js` and `node-notifier` doesn't support `http(s)` for Windows toast notifications.